MSAL/src/instance/MSALOauth2ProviderFactory.m

```diff
@@ -25,34 +25,44 @@
 //
 //------------------------------------------------------------------------------
 
-#import "MSALOauth2FactoryProducer.h"
-#import "MSIDOauth2Factory.h"
-#import "MSIDB2CAuthority.h"
-#import "MSIDAADAuthority.h"
-#import "MSIDAADV2Oauth2Factory.h"
-#import "MSIDB2COauth2Factory.h"
+#import "MSALOauth2ProviderFactory.h"
+#import "MSALB2CAuthority.h"
+#import "MSALAADAuthority.h"
+#import "MSALADFSAuthority.h"
+#import "MSALB2COauth2Provider.h"
+#import "MSALAADOauth2Provider.h"
+#import "MSALADFSAuthority.h"
 
-@implementation MSALOauth2FactoryProducer
+@implementation MSALOauth2ProviderFactory
 
-+ (MSIDOauth2Factory *)msidOauth2FactoryForAuthority:(NSURL *)authority
-                                             context:(id<MSIDRequestContext>)context
-                                               error:(NSError **)error
++ (MSALOauth2Provider *)oauthProviderForAuthority:(MSALAuthority *)authority
+                                          context:(id<MSIDRequestContext>)context
+                                            error:(NSError **)error
 {
     if (!authority)
     {
         MSIDFillAndLogError(error, MSIDErrorInvalidDeveloperParameter, @"Provided authority url is nil.", nil);
-
+        
         return nil;
     }
-
-    if ([MSIDB2CAuthority isAuthorityFormatValid:authority context:context error:nil])
+    
+    if ([authority isKindOfClass:[MSALB2CAuthority class]])
     {
-        return [MSIDB2COauth2Factory new];
+        return [MSALB2COauth2Provider new];
     }
-
-    // Create AAD v2 factory for everything else, but in future we might want to further separate this out
+    else if ([authority isKindOfClass:[MSALAADAuthority class]])
+    {
+        return [MSALAADOauth2Provider new];
+    }
+    else if ([authority isKindOfClass:[MSALADFSAuthority class]])
+    {
+        NSAssert(NO, @"ADFS not implemented in MSAL yet");
```

Comment:
We have an error for unsupported feature and we can bubble that up as a response, no?

Reply:
Fixed